### ISSUE TITLE
[Suggestion] Additional modifiers

### ISSUE THREAD

**Issue by Degonsky** [TheIllusiveC4/ConstructsArmory#162]
Loved this mod so far, especially fact, that it has unique approach to some of the effects :)

While playing, had some ideas popped in mind for modifiers:

1.Weakening
Enchanted golden apple imbued with potion of weakness II makes armor apply weakness I to every meele attacker for 6 seconds with 25% chance, consuming 1 durability when that happends (from 1 part of armor, if wearer has trait on several of them)

2. Shedding
Falling apart armor seems like not a good idea, but if it makes enemies chew piece of metal instead of your head, why not? 

Golden hoe with 2 prismarine shards, large golden plate and mending moss shoud do the trick.

Has 3 levels, can be applied to any armor piece, takes 1 modifier slot for all levels. That modifier makes armor lose point of durability every 20 seconds, with a chance to gain half absorption heart (20% chance for level). Every level increases max amount of said hearts, and decrease proc time by 3 seconds. When gaining absorption point, armor doesnt lose durability. That makes maxing out the trait more efficient but proc faster (with bad luck can drain some durability)

3. Phased
Broken armor quite useless, isnt it? With means to mend it, but not completely repair it it only logical that you need a way to secure it. 

Applying chorus flower, 2 end rods and ender eye with water bucket makes armor stop protecting you when you have less than 50+10% of durability.

Ofc, you can make any changes you see fit, your mod after all :D

**Comment by HOLY-SHIFT** [comment]
also i want to give idea about armor dying. I think modifier-like system should do the trick.
For exemple add a 3 new type of items (little paint buckets or something like this)
1st type is 1st color, 2nd is 2nd, and 3rd is 3rd
each paint bucket can be colored simply placing it with vanila dye in any workbench
so for making green/red/blue armor pice you shall add green 1st layer paint to armor, then red 2nd layer, and blue 3rd layer.
thats it you now have your shiny green/red/blue chestplate, helmet, etc.
hope it would be useful for you

**Comment by Degonsky** [comment]
@HOLY-SHIFT
Author mentioned that there were problems with implementing dyeing, so he discarded that idea untill he figures a good way to implement this. I would like to see it too, but I dont think it will be implemented soon :)

**Comment by HOLY-SHIFT** [comment]
@Degonsky
Yeah, i know it, but i think that my idea is quite good because its dont need too many additional coding. as i mentioned dyes will be modifier-like so TheIllusiveC4 just need to add 3 modifiers with their variations, and make them change how armor look just like redstone, diamond, emerald, etc. does, and the only difference between my dye idea and actual modifier is that dyes will have variations depending on vanila dye colors

**Comment by TheIllusiveC4** [comment]
@HOLY-SHIFT 
> the only difference between my dye idea and actual modifier is that dyes will have variations depending on vanila dye colors

This is more difficult to implement than it sounds. The other modifiers work because they don't need to change colors, they're flat textures. To try and make dyeing possible would require me to revise the entire model baking system.

**Comment by HOLY-SHIFT** [comment]
@TheIllusiveC4 
Well i thought this will work because in my idea dyes not changing color directly, and just cover specified part of the armor. anyway i'm not moder, so i cant argue with you.

By the way here is few of my ideas how this can be realized:
1. add 3 unnecesary slots to armor forge for each armor piece and additional armor parts/dyes to use in these slots (sounds hard to do it this way)
2. add a "dying station" where you have 4 slots: 1 for armor and 3 for dyes
3. make dying process similar to boltcore craft in TC (place armor on casting table, smelt dye, and "cast" a dyed armor) but if you make it this way armor color will be solid (it can be fixed by "3 ingots of dye for  main color, 2 ingots of dye for second color, 1 ingot of dye for other color" or by smelting specified dye for main, second, and other armor color) not sure how hard to make it this way though...

**Comment by mauj3006** [comment]
gostaria que pude-se alterar a quantidades de modificações que podem ser colocadas na  mesma parte da armadura  nas configurações dos mods ou na pasta config exemplo: colocar na calça o modificador de resistência ao fogo e resistência a projeteis 


I would like to be able to change the amount of modifications that can be placed on the same part of the armor in the mods settings or in the config folder example: put the fire resistance and projectile resistance modifier in the pants (
translated from google :P)
